subscription-engine/kafka-ws/app/Redis/RedisManager.js

```diff
@@ -21,6 +22,22 @@ class RedisManager {
     // TODO: Implement "retry_strategy"
     this.redisClient = redis.createClient(this.config);
     logger.info('RedisManager singleton creation complete!');
+
+    const stateService = 'redis';
+    StateManager.registerService(stateService);
+    this.redisClient.on('connect', () => StateManager.signalReady(stateService));
```

Comment:
Is it enough to check only the `connect` and `end` events to the health check?

Reply:
on redis there this is the list of the event available on redis are [ "ready", "connect", "reconnecting", "error", "end", "warining"]

I think the "connect" and "end" events are the ones that represents the actual state of the service.

The documentation can be found here: https://www.npmjs.com/package/redis
If you have any suggestions, they are welcome, the component is subject to changes.

Reply:
Can this be added or not needed?
`this.redisClient.on('reconnecting', () => StateManager.signalNotReady(stateService));`